File: python/bob/bob.py
class Bob:
	def hey(self, sentence):
		# if sentence is empty or whitespace
		if not sentence:
			return 'Fine. Be that way!'
		# if sentence ends with ?
		if sentence[len(sentence)-1] == '?':
			return 'Sure.'
		# if sentence is all uppercase
		if sentence == sentence.upper():
			return 'Woah, chill out!'
		# else return default
		return 'Whatever.'

File: javascript/bob/bob.js
var Bob = module.exports = function() {
	this.hey = function(words) {
		if(words.trim() === '') {
			return "Fine. Be that way!";
		}
		if(words === words.toUpperCase() && words.toUpperCase() != words.toLowerCase()) {
			return "Woah, chill out!";
		}
		if(words[words.length - 1] === '?') {
			return "Sure.";
		}
		return "Whatever.";
	}
};